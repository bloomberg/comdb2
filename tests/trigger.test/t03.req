create table t3 (i int)$$

create procedure t3 version 'test' {
local function main(event)
    db:exec("update t3 set i = i + 1")
    return 0
end}$$

create lua trigger t3 on (table t3 for insert)

insert into t3 values(0)

-- give system time to start trigger
select sleep(3)

-- observe i:0 -> i:1
select * from t3 order by i

create procedure t3 version 'test' {
local function main(event)
    db:exec("update t3 set i = i * 10")
    return 0
end}$$

-- new event will cause system to detect that lua version has changed
insert into t3 values(10)

-- give system time to re-run trigger
select sleep(3)

-- observe i:1 -> i:10, and i:10 -> i:100
select * from t3 order by i
