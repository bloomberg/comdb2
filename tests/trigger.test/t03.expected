(version='test')
(rows inserted=1)
(sleep(3)=3)
(i=1)
(version='test')
(rows inserted=1)
(sleep(3)=3)
(i=10)
(i=100)
